// @sfpy/node-sdk/dist/utils/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionPauseBehavior = exports.SECRET_KEY_SANDBOX = exports.SECRET_KEY_DEVELOPMENT = exports.SECRET_KEY_LOCAL = exports.Environment = exports.CHECKOUT_DEVELOPMENT = exports.CHECKOUT_SANDBOX = exports.CHECKOUT_PRODUCTION = exports.API_URL_DEVELOPMENT = exports.API_URL_SANDBOX = exports.API_URL_PRODUCTION = void 0;
exports.API_URL_PRODUCTION = 'https://api.getsafepay.com';
exports.API_URL_SANDBOX = 'https://sandbox.api.getsafepay.com';
exports.API_URL_DEVELOPMENT = 'https://dev.api.getsafepay.com';
exports.CHECKOUT_PRODUCTION = 'https://getsafepay.com/checkout';
exports.CHECKOUT_SANDBOX = 'https://sandbox.api.getsafepay.com/checkout';
exports.CHECKOUT_DEVELOPMENT = 'https://dev.api.getsafepay.com/checkout';
var Environment;
(function (Environment) {
    Environment["Production"] = "production";
    Environment["Sandbox"] = "sandbox";
    Environment["Development"] = "development";
})(Environment = exports.Environment || (exports.Environment = {}));
exports.SECRET_KEY_LOCAL = 'd23c27c21ee6815d49110ee195b965a57b1ae9a94135a15bb446f6aab499bfdc';
exports.SECRET_KEY_DEVELOPMENT = 'b2fdcf2a147ca808a08fddc7ee5145163f07db62d88d8834733164b819dce73d';
exports.SECRET_KEY_SANDBOX = 'd928d9b1d9e487957c6658be637621bbb104d64b6b436e86714c2ff3712bd825';
var SubscriptionPauseBehavior;
(function (SubscriptionPauseBehavior) {
    SubscriptionPauseBehavior["UnspecifiedCollectionbBehavior"] = "UNSPECIFIED_PAYMENT_COLLECTION_BEHAVIOR";
    SubscriptionPauseBehavior["KeepAsReady"] = "KEEP_AS_READY";
    SubscriptionPauseBehavior["MarkUncollectible"] = "MARK_UNCOLLECTIBLE";
    SubscriptionPauseBehavior["MarkVoid"] = "MARK_VOID";
    SubscriptionPauseBehavior["Unrecgonized"] = "UNRECOGNIZED";
})(SubscriptionPauseBehavior = exports.SubscriptionPauseBehavior || (exports.SubscriptionPauseBehavior = {}));
