// @sfpy/node-sdk/dist/types/subscription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlanType = exports.PlanInterval = void 0;
var SubscriptionStatus;
(function (SubscriptionStatus) {
    SubscriptionStatus["NoSubscriptionStatus"] = "NONE_SUBSCRIPTION_STATUS";
    SubscriptionStatus["Active"] = "ACTIVE";
    SubscriptionStatus["PastDue"] = "PAST_DUE";
    SubscriptionStatus["Unpaid"] = "UNPAID";
    SubscriptionStatus["Canceled"] = "CANCELED";
    SubscriptionStatus["Incomplete"] = "INCOMPLETE";
    SubscriptionStatus["Incomplete_Expired"] = "INCOMPLETE_EXPIRED";
    SubscriptionStatus["Trailing"] = "TRAILING";
    SubscriptionStatus["All"] = "ALL";
    SubscriptionStatus["Ended"] = "ENDED";
    SubscriptionStatus["Paused"] = "PAUSED";
    SubscriptionStatus["Unrecognized"] = "UNRECOGNIZED";
})(SubscriptionStatus || (SubscriptionStatus = {}));
var PlanInterval;
(function (PlanInterval) {
    PlanInterval["UnspecifiedInterval"] = "UNSPECIFIED_INTERVAL";
    PlanInterval["Day"] = "DAY";
    PlanInterval["Week"] = "WEEK";
    PlanInterval["Month"] = "MONTH";
    PlanInterval["Year"] = "YEAR";
    PlanInterval["Minute"] = "MINUTE_TEST_INTERVAL";
    PlanInterval["Unrecognized"] = "UNRECOGNIZED";
})(PlanInterval = exports.PlanInterval || (exports.PlanInterval = {}));
var PlanType;
(function (PlanType) {
    PlanType["Unspecified"] = "UNSPECIFIED";
    PlanType["OneTime"] = "ONE_TIME";
    PlanType["Recurring"] = "RECURRING";
    PlanType["Unrecognized"] = "UNRECOGNIZED";
})(PlanType = exports.PlanType || (exports.PlanType = {}));
